concent_api/api-integration-force-report-computed-task-test.py

```diff
@@ -40,8 +39,9 @@ def force_report_computed_task(task_to_compute):
 
 def ack_report_computed_task(task_to_compute):
     ack_report_computed_task = AckReportComputedTask()
-    ack_report_computed_task.report_computed_task = ReportComputedTask()
+    ack_report_computed_task.report_computed_task = ReportComputedTaskFactory()
     ack_report_computed_task.report_computed_task.task_to_compute = task_to_compute
+    sign_message(ack_report_computed_task.report_computed_task, PROVIDER_PRIVATE_KEY)
```

Comment:
golem-messages has added `EXPECTED_OWNERS` to messages a while ago. We should use it to avoid bugs like this.

It would really be much harder to make a mistake like this if signing was followed by something like:
``` python
sign_message(ack_report_computed_task.report_computed_task, PROVIDER_PRIVATE_KEY)
assert TaskMessage.OWNER_CHOICES.requestor in ack_report_computed_task.EXPECTED_OWNERS
```

Yeah, the assert could always go out of sync with the line above it (like in the example above) but at least someone reading the code can be sure that the assert is correct and spot and fix the inconsistency. As it is now, there's really no way to easily spot bad key usage. You have to analyze every case.

Reply:
:+1: So I will also update this in view and other tests (change to requestor).